src/libstd/thread/local.rs

```diff
@@ -457,6 +457,7 @@ pub mod fast {
         }
     }
 
+    #[allow(improper_ctypes)]
```

Comment:
These two `allow`s would not be necessary with more careful handling of generics, right? These function signatures are FFI-safe for all substitutions.